src/pages/Dashboard/Markets/SupplyMarket.tsx

```diff
@@ -76,23 +98,85 @@ export const SupplyMarketUi: React.FC<ISupplyMarketUiProps> = ({
           orderBy: 'apy',
           orderDirection: 'asc',
         }}
+        rowOnClick={rowOnClick}
+        rowKeyIndex={0}
+      />
+      <CollateralConfirmModal
+        asset={confirmCollateral}
+        handleClose={() => setConfirmCollateral(undefined)}
       />
     </div>
   );
 };
 
 const SupplyMarket: React.FC = () => {
-  // Format fetched data into borrow assets
-  // @TODO: fetch actual data
-  const assets = require('./mocks') // eslint-disable-line
-    .assetData // Filter out tokens we don't support (this could happen if a new token was
-    // introduced within the smart contracts and we didn't update our frontend
-    // config)
-    .filter(
-      (asset: Asset) => !Object.prototype.hasOwnProperty.call(CONTRACT_TOKEN_ADDRESS, asset.symbol),
-    );
+  const { account = '' } = useWeb3Account();
+  const userMarketInfo = useUserMarketInfo({ account });
+  const [confirmCollateral, setConfirmCollateral] = useState<Asset | undefined>(undefined);
+  const { mutate: enterMarkets } = useEnterMarkets({
+    onSuccess: () => setConfirmCollateral(undefined),
+    onError: error => {
+      setConfirmCollateral(undefined);
+      throw error;
+    },
+  });
+  const { mutate: exitMarkets } = useExitMarket({
+    onSuccess: () => setConfirmCollateral(undefined),
+    onError: error => {
+      setConfirmCollateral(undefined);
+      throw error;
+    },
+  });
+  const comptrollerContract = useComptroller();
+  const toggleAssetCollateral = (asset: Asset) => {
+    if (!account) {
+      throw new ToastError(
+        'Account Required',
+        'Please connect your wallet to set an asset as collateral.',
+      );
+    } else if (!asset || !asset.borrowBalance.isZero()) {
+      throw new ToastError(
+        'Collateral Required',
+        'Please repay all borrowed assets or set other assets as collateral.',
+      );
+    } else if (!asset.collateral) {
+      try {
+        setConfirmCollateral(asset);
+        enterMarkets({ comptrollerContract, vtokenAddresses: [asset.vtokenAddress], account });
+      } catch (error) {
+        console.log('enter markets error :>> ', error);
+        throw new ToastError(
+          'Collateral Enable Error',
+          `There was a problem enabeling collateral ${asset.name}. Please try again.`,
+        );
+      }
+    } else if (+asset.hypotheticalLiquidity['1'] > 0 || +asset.hypotheticalLiquidity['2'] === 0) {
+      try {
+        setConfirmCollateral(asset);
+        exitMarkets({ comptrollerContract, vtokenAddress: asset.vtokenAddress, account });
+      } catch (error) {
+        throw new ToastError(
+          'Collateral Disable Error',
+          `There was a problem disabeling collateral ${asset.name}. Please try again.`,
+        );
```

Comment:
This case can't happen, `exitMarkets` doesn't throw. You'll have to handle errors through the `onError` callback of `useExitMarket`

Reply:
ExitMarkets does throw. If you send the wrong arguments it will throw an error